adapters/repos/db/queue/queue.go

```diff
@@ -0,0 +1,1063 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package queue
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// defaultChunkSize is the maximum size of each chunk file. Set to 100MB.
+	defaultChunkSize = 100 * 1024 * 1024
+
+	// chunkFileFmt is the format string for the chunk files,
+	chunkFileFmt = "chunk-%d.bin"
+
+	magicHeader = "WV8Q"
```

Comment:
what is a magic header?

Reply:
it's a series of 4 bytes that indicate what kind of file this is. It is commonly used to detect the type of the binary file.
https://en.wikipedia.org/wiki/List_of_file_signatures
In this case it's 3 bytes for Weaviate (WV8) and one for the queue (Q). The choice of bytes is completely arbitrary.